Firestore/Source/API/FIRQuery.mm

```diff
@@ -348,7 +349,13 @@ - (FIRQuery *)queryOrderedByFieldPath:(FIRFieldPath *)fieldPath descending:(BOOL
 }
 
 - (FIRQuery *)queryLimitedTo:(NSInteger)limit {
-  return Wrap(_query.Limit(static_cast<int64_t>(limit)));
+  int32_t internalLimit;
+  if (limit == NSNotFound || limit > std::numeric_limits<int32_t>::max()) {
```

Comment:
One question this raises is how to handle values that exceed the bounds of a 31-bit limit (negative values are disallowed but the value is signed). I've currently implemented this to saturate to no limit, though the alternative would be to reject values that were too large.

Reply:
Yeah, I don't care strongly since having a limit > 2^31 is pretty non-sensical.  So whatever is easy to implement is fine with me.